### ISSUE TITLE
TypeError: Unable to resolve type 'size_t'

### ISSUE THREAD

**Issue by johncadengo** [ffi/ffi#841]
I'm on Apple Silicon, on Mac OS 11 Big Sur. I was able to get the gem to install successfully (with no errors) but when I run a simple command like rake, I will get this error:

```
$ rake secret
/opt/homebrew/lib/ruby/gems/2.7.0/gems/json-1.8.5/lib/json/common.rb:155: warning: Using the last argument as keyword parameters is deprecated
/opt/homebrew/lib/ruby/gems/2.7.0/gems/mime-types-3.3/lib/mime/types/logger.rb:30: warning: `_1' is reserved for numbered parameter; consider another name
/opt/homebrew/lib/ruby/gems/2.7.0/gems/mime-types-3.3/lib/mime/types/logger.rb:30: warning: `_2' is reserved for numbered parameter; consider another name
/opt/homebrew/lib/ruby/gems/2.7.0/gems/mime-types-3.3/lib/mime/types/logger.rb:30: warning: `_3' is reserved for numbered parameter; consider another name
/opt/homebrew/lib/ruby/gems/2.7.0/gems/tzinfo-1.2.5/lib/tzinfo/ruby_core_support.rb:142: warning: Using the last argument as keyword parameters is deprecated
/opt/homebrew/lib/ruby/gems/2.7.0/gems/tzinfo-1.2.5/lib/tzinfo/ruby_core_support.rb:142: warning: Using the last argument as keyword parameters is deprecated
/opt/homebrew/lib/ruby/gems/2.7.0/gems/json-1.8.5/lib/json/common.rb:155: warning: Using the last argument as keyword parameters is deprecated
/opt/homebrew/lib/ruby/gems/2.7.0/gems/json-1.8.5/lib/json/common.rb:155: warning: Using the last argument as keyword parameters is deprecated
/opt/homebrew/lib/ruby/gems/2.7.0/gems/json-1.8.5/lib/json/common.rb:155: warning: Using the last argument as keyword parameters is deprecated
rake aborted!
TypeError: unable to resolve type 'size_t'
/opt/homebrew/lib/ruby/gems/2.7.0/gems/ffi-1.13.1/lib/ffi/types.rb:69:in `find_type'
/opt/homebrew/lib/ruby/gems/2.7.0/gems/ffi-1.13.1/lib/ffi/library.rb:589:in `find_type'
/opt/homebrew/lib/ruby/gems/2.7.0/gems/ffi-1.13.1/lib/ffi/struct.rb:277:in `find_type'
/opt/homebrew/lib/ruby/gems/2.7.0/gems/ffi-1.13.1/lib/ffi/struct.rb:271:in `find_field_type'
/opt/homebrew/lib/ruby/gems/2.7.0/gems/ffi-1.13.1/lib/ffi/struct.rb:311:in `array_layout'
/opt/homebrew/lib/ruby/gems/2.7.0/gems/ffi-1.13.1/lib/ffi/struct.rb:217:in `layout'
/opt/homebrew/lib/ruby/gems/2.7.0/gems/sassc-2.4.0/lib/sassc/native/sass_value.rb:53:in `<class:SassList>'
/opt/homebrew/lib/ruby/gems/2.7.0/gems/sassc-2.4.0/lib/sassc/native/sass_value.rb:52:in `<module:Native>'
/opt/homebrew/lib/ruby/gems/2.7.0/gems/sassc-2.4.0/lib/sassc/native/sass_value.rb:4:in `<module:SassC>'
/opt/homebrew/lib/ruby/gems/2.7.0/gems/sassc-2.4.0/lib/sassc/native/sass_value.rb:3:in `<top (required)>'
/opt/homebrew/lib/ruby/gems/2.7.0/gems/sassc-2.4.0/lib/sassc/native.rb:16:in `require_relative'
/opt/homebrew/lib/ruby/gems/2.7.0/gems/sassc-2.4.0/lib/sassc/native.rb:16:in `<module:Native>'
/opt/homebrew/lib/ruby/gems/2.7.0/gems/sassc-2.4.0/lib/sassc/native.rb:6:in `<module:SassC>'
/opt/homebrew/lib/ruby/gems/2.7.0/gems/sassc-2.4.0/lib/sassc/native.rb:5:in `<top (required)>'
/opt/homebrew/lib/ruby/gems/2.7.0/gems/sassc-2.4.0/lib/sassc.rb:31:in `require_relative'
/opt/homebrew/lib/ruby/gems/2.7.0/gems/sassc-2.4.0/lib/sassc.rb:31:in `<top (required)>'
/opt/homebrew/lib/ruby/gems/2.7.0/gems/activesupport-5.2.3/lib/active_support/dependencies.rb:291:in `require'
/opt/homebrew/lib/ruby/gems/2.7.0/gems/activesupport-5.2.3/lib/active_support/dependencies.rb:291:in `block in require'
/opt/homebrew/lib/ruby/gems/2.7.0/gems/activesupport-5.2.3/lib/active_support/dependencies.rb:257:in `load_dependency'
/opt/homebrew/lib/ruby/gems/2.7.0/gems/activesupport-5.2.3/lib/active_support/dependencies.rb:291:in `require'
/opt/homebrew/lib/ruby/gems/2.7.0/gems/sassc-rails-2.1.2/lib/sassc/rails.rb:5:in `<top (required)>'
/opt/homebrew/lib/ruby/gems/2.7.0/gems/sassc-rails-2.1.2/lib/sassc-rails.rb:3:in `require_relative'
/opt/homebrew/lib/ruby/gems/2.7.0/gems/sassc-rails-2.1.2/lib/sassc-rails.rb:3:in `<top (required)>'
/opt/homebrew/lib/ruby/gems/2.7.0/gems/activesupport-5.2.3/lib/active_support/dependencies.rb:291:in `require'
/opt/homebrew/lib/ruby/gems/2.7.0/gems/activesupport-5.2.3/lib/active_support/dependencies.rb:291:in `block in require'
/opt/homebrew/lib/ruby/gems/2.7.0/gems/activesupport-5.2.3/lib/active_support/dependencies.rb:257:in `load_dependency'
/opt/homebrew/lib/ruby/gems/2.7.0/gems/activesupport-5.2.3/lib/active_support/dependencies.rb:291:in `require'
/opt/homebrew/lib/ruby/gems/2.7.0/gems/thredded-0.16.16/lib/thredded.rb:32:in `<top (required)>'
/opt/homebrew/lib/ruby/gems/2.7.0/gems/bundler-2.1.2/lib/bundler/runtime.rb:74:in `require'
/opt/homebrew/lib/ruby/gems/2.7.0/gems/bundler-2.1.2/lib/bundler/runtime.rb:74:in `block (2 levels) in require'
/opt/homebrew/lib/ruby/gems/2.7.0/gems/bundler-2.1.2/lib/bundler/runtime.rb:69:in `each'
/opt/homebrew/lib/ruby/gems/2.7.0/gems/bundler-2.1.2/lib/bundler/runtime.rb:69:in `block in require'
/opt/homebrew/lib/ruby/gems/2.7.0/gems/bundler-2.1.2/lib/bundler/runtime.rb:58:in `each'
/opt/homebrew/lib/ruby/gems/2.7.0/gems/bundler-2.1.2/lib/bundler/runtime.rb:58:in `require'
/opt/homebrew/lib/ruby/gems/2.7.0/gems/bundler-2.1.2/lib/bundler.rb:174:in `require'
(See full trace by running task with --trace)
```

This is happening in a typical Rails app.

**Comment by hcatlin** [comment]
Can confirm this is happening for me too!

**Comment by yoyostile** [comment]
There is a PR pending here https://github.com/ffi/ffi/pull/801. 
You can check out his branch and build ffi yourself, if you depend on a quick fix. Alternatively go down the usual `arch -x86_64` route and temporarily use rosetta for your rails apps.

**Comment by hcatlin** [comment]
Appreciate all of your work on this! Was just confirming in case it wasn't clear. Trying to be helpful... and thank you all so much for what you do! <3

**Comment by hcatlin** [comment]
I can confirm that the code merged into #801 removed this error and now see #845 instead! We're getting there!

**Comment by rgaufman** [comment]
I'm seeing this too, any ETA for a new gem release? :)

**Comment by felixbuenemann** [comment]
@hcatlin I'm not seeing any crashes for sassc 2.0.1 with Ruby 2.6.6 / 2.7.2 arm64 and ffi master @ a7f4c95 using either the bundled libffi 2.7.2, macOS default or Homebrew libffi 3.3 on a 2020 M1 MacBook Air running macOS 11.1 (20C69) and Xcode / Xcode CLI 12.3.

I agree that a new release would be great.

**Comment by kallewesterling** [comment]
Adding to this - trying to install `jekyll` on M1 Mac mini, and keep running into the same issue.

**Comment by larskanis** [comment]
Compatibility to arm64-darwin is fixed in ffi-1.14.0.
